### DISCUSSION TITLE
How to *only* read first `n` pages of Word Doc or PDF without loading every pages?

### DISCUSSION THREAD

**Discussion by AayushSameerShah** [Unstructured-IO/unstructured#2214] [Q&A]
# Question - 1️⃣
> ### ⁉
>
> Is that possible that I want the user to limit spamming by uploading a novel as a word doc in my server *(yes I can set the MB size but, you know)* and I know that typically under 3 pages of word doc all information would be covered, so how can I set the number of pages to read only while reading the word doc or PDF **without reading all pages** and then filtering out.

# Question - 2️⃣
> ### ⁉
>
> Is it possible to know the number of pages beforehand? I mean just to get the `len()` kind of feature? Without even partitioning the file? Can I get the number of pages?

Please help, thanks!

**Comment by ericfeunekes** [comment]
I would suggest just doing it yourself. You'll have. The libraries you need installed since they are dependencies of this library. 

Then just check the page count or whatever else you need, create a copy of the file in memory with only three pages and pass that to unstructured. 

Same for length. Load the file and get the number of pages.
